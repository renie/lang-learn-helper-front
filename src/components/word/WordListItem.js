import React from 'react'

import './style/wordListItem.css'

const WordListItem = ({word={}, MeaningListComponent, key}) => {
    return (
        <li className="card wordListItem">
            <header className="card-header">
                <p className="title">
                    {word.name}
                </p>
            </header>
            <div className="card-content">
                <div className="content">
                    <p className="subtitle">Meanings:</p>
                    {word.meanings && <MeaningListComponent meanings={word.meanings} />}
                </div>
            </div>
        </li>
    )
}

export default WordListItem
