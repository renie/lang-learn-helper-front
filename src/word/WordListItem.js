import React from 'react'

import './wordListItem.css'

const WordListItem = ({word={}, MeaningList}) => {
    return (
        <li className="card wordListItem">
            <header className="card-header">
                <p className="title">
                    {word.name}
                </p>
            </header>
            <div className="card-content">
                <div className="content">
                    <p className="subtitle">Meanings:</p>
                    {word.meanings && <MeaningList meanings={word.meanings} />}
                </div>
            </div>
        </li>
    )
}

export default WordListItem
