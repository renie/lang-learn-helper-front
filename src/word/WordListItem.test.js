import React from 'react'
import ReactDOM from 'react-dom'
import renderer from 'react-test-renderer'

import WordListItem from './WordListItem'

const MeaningList = ({meanings}) => `${meanings[0].wordClass} - ${meanings[0].id}`

it('renders correctly empty', () => {
    const tree = renderer
                    .create(<WordListItem 
                                word={undefined}
                                MeaningListComponent={MeaningList}/>)
                    .toJSON()

    expect(tree).toMatchSnapshot()
})

it('renders correctly not empty', () => {
 
    const word = {
        "name":"used",
        "meanings":[{
            "id": 1,
            "wordClass": "verb",
            "example": "He used my pen.",
            "description": "Verb use in past tense."
        },
        {
            "id":2,
            "wordClass":"adjective",
            "example":"My pen is used.",
            "description":"Means the object is not new anymore."
        }],
        "_id":"BcNdVZy038MWxqrg"
    }

    const tree = renderer
                    .create(<WordListItem
                                word={word}
                                MeaningListComponent={MeaningList}/>)
                    .toJSON()

  expect(tree).toMatchSnapshot()
})
