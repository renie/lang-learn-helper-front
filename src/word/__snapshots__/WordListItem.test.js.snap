// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly empty 1`] = `
<li
  className="card wordListItem"
>
  <header
    className="card-header"
  >
    <p
      className="title"
    />
  </header>
  <div
    className="card-content"
  >
    <div
      className="content"
    >
      <p
        className="subtitle"
      >
        Meanings:
      </p>
    </div>
  </div>
</li>
`;

exports[`renders correctly not empty 1`] = `
<li
  className="card wordListItem"
>
  <header
    className="card-header"
  >
    <p
      className="title"
    >
      used
    </p>
  </header>
  <div
    className="card-content"
  >
    <div
      className="content"
    >
      <p
        className="subtitle"
      >
        Meanings:
      </p>
      verb - 1
    </div>
  </div>
</li>
`;
